storage/txcache/maps/bucketSortedMapItem.go

```diff
@@ -1,12 +1,32 @@
 package maps
 
-import "github.com/ElrondNetwork/elrond-go/core/atomic"
+import (
+	"sync/atomic"
+)
 
-// BucketSortedMapItem is
+// BucketSortedMapItem defines an item of the bucket sorted map
 type BucketSortedMapItem interface {
 	GetKey() string
 	ComputeScore() uint32
-	GetScoreChunk() *MapChunk
-	SetScoreChunk(*MapChunk)
-	ScoreChangeInProgressFlag() *atomic.Flag
+	ScoreChunk() *MapChunkPointer
+}
+
+// MapChunkPointer is an atomically-mutable pointer to a MapChunk
+type MapChunkPointer struct {
```

Comment:
So, you basically created a concurrent "safe" structure just to hold the pointer? Why didn't you simply add a mutex on GetScoreChunk and SetScoreChunk? The project uses a lot the later version.

Reply:
Indeed. Will fix to use a mutex (and also make tiny benchmark for, like, 10 routines with 1k writes each - SetScoreChunk - with mutex vs. atomic mutable pointer).

Reply:
Fixed :)